jwt/api_jwk.py

```diff
@@ -47,9 +47,13 @@ def __init__(self, jwk_data, algorithm=None):
         self.Algorithm = self._algorithms.get(algorithm)
 
         if not self.Algorithm:
-            raise PyJWKError("Unable to find a algorithm for key: %s" % self._jwk_data)
+            import logging
 
-        self.key = self.Algorithm.from_jwk(self._jwk_data)
+            self.key = None
+            logger = logging.getLogger(__name__)
+            logger.warning("Do not support algorithm %s", algorithm)
+        else:
+            self.key = self.Algorithm.from_jwk(self._jwk_data)
```

Comment:
I think this might have security implications, as users of this class likely expect the algorithm in the JWK data to be valid.